src/scripts/main.js

```diff
@@ -1,3 +1,43 @@
 'use strict';
 
-// write code here
+function convertToNum(item, key) {
+  return +item[key].replace(/\$|,/g, '') || undefined;
+}
+
+function getEmployees(list) {
+  const resultArray = [];
+  const employeeList = [...list];
+
+  employeeList.map((item, index) => {
+    resultArray.push({
+      ...item.dataset, name: item.innerText,
+    });
+    resultArray[index].salary = convertToNum(resultArray[index], 'salary');
+  });
```

Comment:
do not `push` values to other arrays using map , just create an object on each iteration and SAVE your result of map to variable, like that:
```suggestion
  const employeeList = [...list];

   const resultArray = employeeList.map((item, index) => {
	   return {
	      ...item.dataset,
	      name: item.innerText,
	      salary: convertToNum(item, 'salary')
	    });
  });
```